cdn_lambda/functions/map_to_s3.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+
+from chexus import Client, TableItem
+
+TABLE_NAME = "project-exodus-prototyping"
+TABLE_REGION = "us-east-2"
+
+
+def lambda_handler(event, context):
+    _ = context
+
+    # Get uri from origin request event
+    request = event["Records"][0]["cf"]["request"]
+
+    client = Client()
+
+    # Get DB table items with matching web_uri key
+    search_results = client.search(
+        item=TableItem(web_uri=request["uri"]),
+        table_name=TABLE_NAME,
+        region=TABLE_REGION
+    )
+
+    items = [TableItem(**i) for i in search_results["Items"]
+             if i.attrs["from_date"] <= datetime.utcnow().date()]
+
+    if items:
+        # Sort the items by date
+        items.sort(key=lambda i: i.from_date)
+
+        # Redirect the request uri
+        request["uri"] = items[-1].attrs["object_key"]
```

Comment:
I have a question about logging, which might need some looking into.

If something goes wrong here - e.g. imagine that the item from the DB is missing object_key somehow, so this raises KeyError - do we get meaningful logs somewhere with the full backtrace?  Or do we need to surround the entire thing with an exception handler and a call to a logger?

It's fine to leave this question for later though.

Reply:
No, I do think we'll need to catch things like that here. Our tooling can do everything to ensure it abides by the schema but I think there's still room for invalid content to find it's way onto the DB, and this is that content's first stop.